crates/sqlbuiltins/src/functions/table/json.rs

```diff
@@ -0,0 +1,67 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use datafusion::datasource::TableProvider;
+use datafusion_ext::errors::ExtensionError;
+use datafusion_ext::functions::{FuncParamValue, TableFuncContextProvider};
+use datasources::common::url::{DatasourceUrl, DatasourceUrlType};
+use datasources::json::table::json_streaming_table;
+use datasources::object_store::generic::GenericStoreAccess;
+use protogen::metastore::types::catalog::RuntimePreference;
+
+use crate::functions::table::{table_location_and_opts, TableFunc};
+use crate::functions::{ConstBuiltinFunction, FunctionType};
+
+#[derive(Debug, Clone, Copy, Default)]
+pub struct JsonScan;
+
+impl ConstBuiltinFunction for JsonScan {
+    const NAME: &'static str = "read_json";
+    const DESCRIPTION: &'static str = "Reads one or more JSON files or URLs. Supports globbing.";
+    const EXAMPLE: &'static str = "SELECT * FROM read_json('https://example.com/feed.json')";
+    const FUNCTION_TYPE: FunctionType = FunctionType::TableReturning;
+}
+
+#[async_trait]
+impl TableFunc for JsonScan {
+    fn detect_runtime(
+        &self,
+        args: &[FuncParamValue],
+        _parent: RuntimePreference,
+    ) -> Result<RuntimePreference, ExtensionError> {
```

Comment:
Id expect this to error if you try to provide non matching urls. `read_json(['file://...', 'https://...'])`

Reply:
none of our other data sources have this semantic. 

I agree, though.

Reply:
They did at one point. I remember writing the logic for this. 

If that got removed, we should open up an issue as this is a regression.

Reply:
The object store one does: https://github.com/glaredb/glaredb/blob/c8275d47909a5f5b0380055b5b192510f9b340c7/crates/sqlbuiltins/src/functions/table/object_store.rs#L221-L236

Which calls into this: https://github.com/glaredb/glaredb/blob/c8275d47909a5f5b0380055b5b192510f9b340c7/crates/sqlbuiltins/src/functions/table/object_store.rs#L309-L348